tests/lib/config.js

```diff
@@ -884,6 +884,38 @@ describe("Config", function() {
             assertConfigsEqual(actual, expected);
         });
 
+        it("should extend package sub-configuration with prefix", function() {
+
+            var configDeps = {};
+            configDeps["./config/config-file"] = proxyquire("../../lib/config/config-file", {
+                "@scope/eslint-config/foo": {
+                    rules: {
+                        eqeqeq: 2
+                    }
+                }
+            });
+
+            var StubbedConfig = proxyquire("../../lib/config", configDeps);
+
+            var configPath = path.resolve(__dirname, "../fixtures/config-extends/scoped-package9/.eslintrc"),
+                configHelper = new StubbedConfig({ useEslintrc: true, configFile: configPath, cwd: process.cwd() }),
+                expected = {
+                    rules: { "quotes": [2, "double"], "eqeqeq": 2, "valid-jsdoc": 0 },
+                    env: { "browser": false }
+                };
+
+            // Reason to override this function in this special case is that I don't want it to keep looking into the
+            // chain of parent directories for .eslintrc or package.json file for configs.
+            // If that is allowed then the expected outcome will change based on the machine you are running this test.
+            sinon.stub(configHelper, "findLocalConfigFiles").returns([
+                path.resolve(__dirname, "../fixtures/config-extends/scoped-package9/.eslintrc")
```

Comment:
I've made the adjustments but not yet committed because of the test with the label "should extend package configuration with sub directories". The test appears to assert that when targeting a file, the config should look for the .eslintrc file in the parent directory automatically, which extends another config (hence `useEslintrc` is set to true there). Though, `configFile` is being passed explicitly which feels wrong and will make the test to pass, even if the config is not being loaded correctly. Am I missing something?

The test should be:

```js
        it("should extend package configuration with sub directories", function() {

            var configDeps = {};
            configDeps["./config/config-file"] = proxyquire("../../lib/config/config-file", {
                "eslint-config-foo": {
                    rules: {
                        "eqeqeq": 2
                    }
                }
            });

            var StubbedConfig = proxyquire("../../lib/config", configDeps);

            var configHelper = new StubbedConfig({ useEslintrc: true, cwd: process.cwd() }),
                expected = {
                    rules: { "quotes": [2, "double"], "eqeqeq": 2, "valid-jsdoc": 0 },
                    env: { "browser": false }
                },
                actual = configHelper.getConfig(path.resolve(__dirname, "../fixtures/config-extends/package2/subdir/foo.js"));

            assertConfigsEqual(actual, expected);
        });
```

Also the `root: true` should be added to package2.json.

*Note*: In fact, I think the `root:true` should be added to all fixtures by default, unless there's a specific reason no to.